### ISSUE TITLE
CloudFlare - cfscrape update

### ISSUE THREAD

**Issue by Twoure** [Twoure/KissNetwork.bundle#15]
CloudFlare anti-bot code updated.  Currently updating to latest cfscrape code to fix.

Also comic section is now behind same anti-bot, so updating cache cover code to handle new comic cover thumbs.

**Comment by Twoure** [comment]
Fixed in [v1.2.1](https://github.com/Twoure/KissNetwork.bundle/releases/tag/v1.2.1)

**Comment by ruifung** [comment]
2016-06-05 22:57:48,460 (7f2766ffd700) :  ERROR (DevTools:572) - ********************************************************************************
2016-06-05 22:57:48,461 (7f2766ffd700) :  ERROR (DevTools:573) - ----------CFTest Fail----------
2016-06-05 22:57:48,462 (7f2766ffd700) :  ERROR (DevTools:574) - You need to install a JavaScript Runtime like node.js or equivalent
2016-06-05 22:57:48,462 (7f2766ffd700) :  ERROR (DevTools:575) - Once JavaScript Runtime installed, Restart channel
2016-06-05 22:57:48,463 (7f2766ffd700) :  ERROR (DevTools:576) - stdout: '', stderr: ''
2016-06-05 22:57:48,463 (7f2766ffd700) :  ERROR (DevTools:577) - ********************************************************************************

I do have nodejs installed.

(nodejs invokes well... node js.)

**Comment by Twoure** [comment]
Try restarting PMS?

**Comment by ruifung** [comment]
That was the first thing I tried. Restart PMS, reinstall bundle, wipe its data folders. Anything else I missed?

Heck, isn't that output pretty strange for a exception?

**Comment by Twoure** [comment]
> Heck, isn't that output pretty strange for a exception?

Yes.

> 2016-06-05 22:57:48,463 (7f2766ffd700) : ERROR (DevTools:576) - stdout: '', stderr: ''

This is error dump caught by the exception when the cfscrape test failed.  All the other lines are constants I output when the cfscrape test fails.  

In the past I tried updating `PyExecJS` but had similar issues.  My assumption is the newer version of `PyExecJS` is messing up due to some unmet dependence within your OS.

What OS is you PMS hosted on?  I'll take some time and make a new branch for you to test.

**Comment by Twoure** [comment]
Added javascript runtime engine info to the error within the [dev](https://github.com/Twoure/KissNetwork.bundle/tree/dev) branch.  Download and try the [dev](https://github.com/Twoure/KissNetwork.bundle/tree/dev) branch and let me know what the error says.

You can also try removing the current `execjs` and replace it with the old version [v1.1.0](https://github.com/doloopwhile/PyExecJS/tree/v1.1.0/execjs).  If that works, then please let me know, as-well-as what OS your PMS is hosted on.  Thanks.

**Comment by ruifung** [comment]
```
DEBUG (headers:36) - * Javascript runtime engine = Node.js (V8)
```

OS: Ubuntu 14.04 LTS
NodeJS version: v0.10.25
Python version: 2.7.6
Python3 version: 3.4.3

execjs 1.1.0 works.
All execjs versions from 1.2.0 onwards cause the issue for me.

Confirmed that once I replaced execjs with 1.1.0, everything works (search, browsing, playback).. although is it normal for anime to not have thumbnails?

**Comment by Twoure** [comment]
> Confirmed that once I replaced execjs with 1.1.0, everything works (search, browsing, playback)

Glad it's working for you now, and thank you for the debug info requested.  I see you're running the same OS as I am so that rules out an OS depended error (maybe).  I'll look into reverting `execjs` back to v1.1.0 for the next release.

> is it normal for anime to not have thumbnails?

Yes, to make thumbnails show enable [Cache All Covers Locally](https://github.com/Twoure/KissNetwork.bundle/tree/v1.2.1#cache-all-covers-locally) from within the channel's preferences.  More info as to why, can be found in this know issue [bullet 3](https://github.com/Twoure/KissNetwork.bundle/tree/v1.2.1#anime-cartoon-drama-manga-comics).

**Note:** for Ubuntu make sure the channels `Resources` directory has proper write permissions so the channel can cache the cover images.  Relevant info can be found in [Comment_1077529](https://forums.plex.tv/discussion/comment/1077529/#Comment_1077529).

**Comment by ruifung** [comment]
Well, I do have quite a bit of PPA's installed, so maybe packages from one of them are affecting it. If you have any idea on what to check I can probably get more detailed information.

**Comment by Twoure** [comment]
Trying to add `execjs v1.1.0` as fallback.  Please download and test the dev branch again ([dev download link](https://github.com/Twoure/KissNetwork.bundle/archive/dev.zip)).  Let me know how it goes.  Thanks.

**Comment by ruifung** [comment]
I haven't been checking github for a few days (been buried in a book), I'll test it when I have the time.

And... I'll just checkout the dev branch.

**Comment by Twoure** [comment]
Closing, because original issue is resolved in [v1.2.2](https://github.com/Twoure/KissNetwork.bundle/releases/tag/v1.2.2).
